src/models/annotations/arrow-annotation.ts

```diff
@@ -6,6 +6,10 @@ import { uniqueId } from "../../utilities/js-utils";
 
 export const kArrowAnnotationType = "arrowAnnotation";
 
+export function isArrowAnnotation(annotation: any) {
+  return "type" in annotation && annotation.type === kArrowAnnotationType;
+}
```

Comment:
These type guard functions are generally used to provide additional type information so that TypeScript can make additional type inferences. I think what would make sense here would be to define `IArrowAnnotationSnapshot` below:
```typescript
export interface IArrowAnnotationSnapshot extends SnapshotIn<typeof ArrowAnnotation> {}
```

and then this function becomes:
```suggestion
export function isArrowAnnotationSnapshot(snapshot: any): snapshot is IArrowAnnotationSnapshot {
  return "type" in snapshot && snapshot.type === kArrowAnnotationType;
}
```